docs/resources/troubleshooting.md

```diff
@@ -96,4 +97,25 @@ This error typically indicates that data was not ingested into the **ingestion**
 
 ---
 
+## Deployment error: The resource write operation failed to complete successfully, because it reached terminal provisioning state 'Failed'. (Code: ResourceDeploymentFailure, Target: /subscriptions/subscription/resourceGroups/resourcegroup/providers/Microsoft.Resources/deployments/storage)
+
+If you've deleted FinOps Hubs and are attempting to redeploy it with the same values, including the Managed Identity name, you might encounter the following known issue:
+
+```json
+"code": "RoleAssignmentUpdateNotPermitted",
+"message": "Tenant ID, application ID, principal ID, and scope are not allowed to be updated."
+```
+
+### Resolution Options:
+
+1. **Remove Stale Identity:**
+   - Navigate to "Storage Account >> Access Control IAM" >> "Role assignments."
+   - Identify a role assignment with an "unknown" identity and delete it.
+
+2. **Update Role Assignment Name (GUID):**
+   - Choose a GUID that hasn't been used previously.
+   - Retry the deployment with the updated role assignment name.
```

Comment:
The template doesn't allow you to select the role assignment name. This is part of the problem. I think deleting it is the only option.